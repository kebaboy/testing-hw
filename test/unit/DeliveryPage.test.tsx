// check cur url

import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import { Home } from "../../src/client/pages/Home";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import { basename, renderApp } from "../utils";
import { Application } from "../../src/client/Application";

describe('Страница условий доставки', () => {
    it('должна иметь статическое содержание', () => {
        const {container} = renderApp(<Application/>, [`${basename}/delivery`]);

        expect(container).toMatchSnapshot('Страница условий доставки');
    })

    it('ссылка "Delivery" в навигации должна быть активной', () => {
        const { container } = renderApp(<Application />, [`${basename}/delivery`]);
        const link = screen.getAllByRole('link', { name: /delivery/i })[0];

        // screen.logTestingPlaygroundURL();
        expect(link).toBeInTheDocument();
        expect(link).toHaveClass('active');
    })
})