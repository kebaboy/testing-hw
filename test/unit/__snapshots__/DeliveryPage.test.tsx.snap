// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Страница условий доставки должна иметь статическое содержание: Страница условий доставки 1`] = `
<div>
  <div
    class="Application"
  >
    <nav
      class="navbar navbar-expand-sm navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="Application-Brand navbar-brand"
          href="/hw/store"
        >
          Kogtetochka store
        </a>
        <button
          aria-label="Toggle navigation"
          class="Application-Toggler navbar-toggler"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="Application-Menu collapse navbar-collapse"
        >
          <div
            class="navbar-nav"
          >
            <a
              class="nav-link"
              href="/hw/store/catalog"
            >
              Catalog
            </a>
            <a
              aria-current="page"
              class="active nav-link"
              href="/hw/store/delivery"
            >
              Delivery
            </a>
            <a
              class="nav-link"
              href="/hw/store/contacts"
            >
              Contacts
            </a>
            <a
              class="nav-link"
              href="/hw/store/cart"
            >
              Cart
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container pt-4"
    >
      <div
        class="Delivery"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h1>
              Delivery
            </h1>
            <p>
              Swift and Secure Delivery: Experience the convenience of hassle-free shipping with our scratchers. We understand the excitement of receiving your new cat furniture, so we prioritize swift delivery to your doorstep. Rest assured, your order is handled with care every step of the way, ensuring it arrives safely and securely.
            </p>
            <img
              class="Image w-25 mb-4"
              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
            />
            <p>
              Track Your Package with Ease: Stay informed and in control of your delivery with our easy-to-use tracking system. From the moment your order is placed to the minute it reaches your home, you can monitor its journey in real-time. No more guessing games – know exactly when to expect your package and plan accordingly.
            </p>
            <p>
              Customer Satisfaction Guaranteed: Your satisfaction is our top priority, which is why we go above and beyond to provide exceptional delivery service. If you have any questions or concerns about your shipment, our dedicated customer support team is here to assist you every step of the way. Trust us to deliver not only your scratcher but also peace of mind.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
